Audio Modes: Add full/sfx-only/muted toggle with UI icon support

emna9 opened the pull request:
Addresses issue #1117 (easy toggle in-game sound modes )

This pull request introduces a new audio mode toggle system for Ancient Beast, giving players full control over their sound experience.

#### Available Modes:

- **Full**: Music + SFX  
- **SFX Only**: Sound effects only, no music  
- **Muted**: No audio at all  

#### How to Use:

- Right-click the audio icon in the top-right corner to cycle through the modes.  
- The audio icon updates visually to reflect the current mode.  

#### Fully Tested

- Works in both development and production builds.

#### Modes in Action

**Full Audio Mode**  
![Audio:Full](https://github.com/user-attachments/assets/85460f1e-4acd-4b58-add2-a7831f29260b)

**SFX Only Mode**
![Audio:SFX Only](https://github.com/user-attachments/assets/f7fcd582-409c-498d-8948-08cff87e922d)

**Muted Mode**
![Audio:Muted](https://github.com/user-attachments/assets/732e7df6-59c6-4535-ba3c-8363be896697)

DreadKnight commented:
@emna9 Heya! You've touched some extra files, some with bogus stuff, carefully check out the files.
https://github.com/FreezingMoon/AncientBeast/pull/2681/files

This works pretty well, but the idea is to change the big icon, not to add an extra small one around 🐻
You can drop the " only" for SFX.

emna9 commented:
Thanks for the review 🫡
You're right , I’ll refactor the small icon to replace the big audio icon directly instead.
Also noted! I’ll drop the only from the SFX tooltip label to keep things clean.
PLus I will send you the new screenshots in a few minutes so you can confirm if that’s what you had in mind.

emna9 commented:
@DreadKnight

DreadKnight commented:
@emna9 No need to touch readme.md and game.ts files with bogus edits.
Also, icons wise, the default one for the "full" audio, the one with curves on the right would be the "sfx only".

emna9 commented:
well received , so effects.svg goes for Audio:Full, music.svg goes for SFX and music-off.svg goes for muted ??
Also i did a small change on game.ts that initializes the game's audio system in full mode on load,avoiding any desync between functionality and visuals.

DreadKnight commented on README.md:
```diff
@@ -4,7 +4,7 @@
 [![Discord](https://img.shields.io/discord/154868963132571649?logo=discord&label=Discord&color=5865F2)](https://discord.gg/CtqBsnF85z)
 
-
+o
```
What's this? 🐻

DreadKnight commented:
> @DreadKnight well received , so effects.svg goes for Audio:Full, music.svg goes for SFX and music-off.svg goes for muted ?? Also i did a small change on game.ts that initializes the game's audio system in full mode on load,avoiding any desync between functionality and visuals.

@emna9 Started a review with the weirdness for readme file.

Regarding the icons, you could have all 3 in that audio folder provided you don't break anything; you can rename the files according and get rid of the other unused icons.

emna9 commented:
@DreadKnight Thanks for pointing out the README issue ,that was a mistake I’ll fix it right away

Also just to clarify the icons I used are already in the icons/audio folder

DreadKnight commented:
> @DreadKnight Thanks for pointing out the README issue ,that was a mistake I’ll fix it right away
> 
> Also just to clarify the icons I used are already in the icons/audio folder

The full one is not in there. You can get rid of that audio folder and just move those 2 icons in the upper folder along with the current audio.svg

As for `src/game.ts` it only has a "}" at the end which wasn't missing in the first place afaik, odd...

emna9 commented:
@DreadKnight Just finished cleaning up both README.md and src/game.ts. Everything should be good to go now,let me know if you spot anything else

emna9 commented:
heyy @DreadKnight thanks for the review I removed commented code and uncommented the border styling as you said also i moved the music icons to icons folder renamed them and removed them from audio folder but I didn’t delete the audio folder entirely since it still contains a few other icons that are in use and that wasn’t part of the issue I was handling. should I push my updates now?🐻

DreadKnight commented:
> heyy @DreadKnight thanks for the review I removed commented code and uncommented the border styling as you said also i moved the music icons to icons folder renamed them and removed them from audio folder but I didn’t delete the audio folder entirely since it still contains a few other icons that are in use and that wasn’t part of the issue I was handling. should I push my updates now?🐻

Yeah, push updates. Will review that folder and delete stuff myself, though I'm pretty sure the extra icons there aren't used.

emna9 commented:
@DreadKnight updates are pushed , plus when I tested I found I think back.svg is used in audio folder

DreadKnight commented:
@emna9 I'm pretty sure that icon is not used. All those icons where supposedly picked for some very cheesy retro fullscreen audio player that would have been kinda obnoxious. Current stuff looks pretty neat imo, with a nice vampire-goth vibe to it 🦇

Did more clean-up on you branch, gonna merge 🐻

Don't forget to provide an ERC-20 crypto wallet address for your project token bounty 🪙
https://github.com/FreezingMoon/AncientBeast/blob/master/CONTRIBUTING.md#token-bounties

emna9 commented:
@DreadKnight Thank you for cleaning it up and merging ! Still getting used to the flow, sorry for the small hiccups its my first time contributing in here .
Thanks again for the follow up 🫶 🫶
